src/NzbDrone.Core/DecisionEngine/DownloadDecisionPriorizationService.cs

```diff
@@ -13,18 +13,21 @@ public interface IPrioritizeDownloadDecision
     public class DownloadDecisionPriorizationService : IPrioritizeDownloadDecision
     {
         private readonly IDelayProfileService _delayProfileService;
+        private readonly IIndexerFactory _indexerFactory;
 
-        public DownloadDecisionPriorizationService(IDelayProfileService delayProfileService)
+        public DownloadDecisionPriorizationService(IDelayProfileService delayProfileService, IIndexerFactory indexerFactory)
         {
             _delayProfileService = delayProfileService;
+            _indexerFactory = indexerFactory;
         }
 
         public List<DownloadDecision> PrioritizeDecisions(List<DownloadDecision> decisions)
         {
+            var indexers = _indexerFactory.All();
             return decisions.Where(c => c.RemoteEpisode.Series != null)
                             .GroupBy(c => c.RemoteEpisode.Series.Id, (seriesId, downloadDecisions) =>
                                 {
-                                    return downloadDecisions.OrderByDescending(decision => decision, new DownloadDecisionComparer(_delayProfileService));
+                                    return downloadDecisions.OrderByDescending(decision => decision, new DownloadDecisionComparer(_delayProfileService, indexers));
```

Comment:
Taloth mentioned this to me in our discussion, but hasn't had a chance to mention it here:

"the Comparer should have no knowledge of the Indexer settings, nor of the Indexer Definitions. Meaning the Comparer must be provided either a Dictionary of Indexer ID to Priority, or better should get an interface to a class that will provide it."

Given the comparer isn't using DI I think we'll want to pass in an instance of the class that is responsible for looking up the priority, which can also be responsible for caching the priority for a short period of time (maybe 5 minutes? and possible cleared when indexers are saved?).

Reply:
Agreed, except on the caching part. Looking at other bits of code I don't see any similar caching and I don't think it will make a noticable difference here. 'Premature optimization is the root of all evil' - it will add unnecessary complexity to the code which could contain bugs. We could add the caching part in the future if performance becomes an issue.

Reply:
You were caching before and this seems like a pretty high impact area for caching the priority for a few minutes to drop the DB calls to pretty much 1 per indexer per RSS Sync/search.

`DelayProfileService` and `SceneMappingService` are both in the release processing path that use caching, with `DelayProfileService` wiping the cache when a delay profile is added/updated/deleted.

Reply:
As I'm beginning to implement this I'm strongly feeling that this might be a bit overengineered. Since we moved the priority from the settings object to the IndexerDefinition class the code is simplified, Moving it to a new class seems kind of silly and makes the code less readable (code is spread out). What do you think @Taloth

Reply:
Code is already spread out: DownloadDecisionPriorizationService calls IndexerFactory to get the list of indexers, only to provide the Comparer with that result. The Comparer converts it into a Dictionary and stores that. Then later, the Comparer uses the dictionary to get the IndexerDefinition for a specific indexer id, pulls the Priority property from it and uses that during comparisons.

If you add a separate service (such as `IProvideIndexerPriority`), then the logic becomes localized in that one class. DownloadDecisionPriorizationService no longer has to fetch the list of indexers, and the Comparer won't have to fiddle with IndexerDefinitions either.

If I want to create unittests for the Comparer, I no longer have to create a bunch of fake IndexerDefinitions. Instead I can just mock up the method in IProvideIndexerPriority and return the priority directly.

I get why you think this is overengineering, but it has a specific purpose in mind: To hide the inner workings of one part of the application from another.
To name another example: IIndexerFactory has several methods to provide Enabled, or RssEnabled, or SearchEnabled indexers. It could've just as easily provided an All() method and have the caller filter the result, but that would mean the caller becomes aware of which settings cause an indexer to be 'Enabled' duplicating that particular code everywhere it's needed.
The Enabled logic is of course more complex than Priority, but it's the same principle.

I hope that clarifies our thought process a bit.